File: Python/BOJ/BRONZE/1193_분수찾기.py
n = int(input())                    # 분수 번호 n 입력
line = 1                            # 분수 라인 1부터 시작

while (n > line):                   # n이 line보다 작아질 때까지 반복
    # 입력받은 n을 1씩 늘려가며 빼줌(몇 번째 줄에 몇 번째 숫자인지 구하기)
    n -= line
    line += 1

if line % 2 == 0:                # 짝수 라인이라면
    a = n                           # 분자a는 오름차순
    b = line-n+1                    # 분자b는 내림차순
else:                            # 홀수 라인이라면
    a = line-n+1                    # 분자a는 내림차순
    b = n                           # 분모b는 오름차순

print(a, '/', b, sep='')            # 분수 출력

File: C/BOJ/BRONZE/1193_분수찾기.c
#include <stdio.h>

int main()
{
	int n, diff, top, bottom, line = 0, end = 0;
	// line은 대각선으로 그었을 때 짝수 또는 홀수 번째에 있는지 판별하는 변수
	// end는 해당 라인의 가장 끝의 번째(1, 3, 6, 10, ...)를 나타내는 변수

	scanf("%d", &n);

	while (n > end) {
		line += 1;
		end += line;
	}

	diff = end - n;

	if (line % 2 == 0) {
		top = line - diff;
		bottom = diff + 1;
	}
	else {
		top = diff + 1;
		bottom = line - diff;
	}

	printf("%d/%d", top, bottom);
}
